pkgs/build-support/testers/expect-failure.sh

```diff
@@ -35,7 +35,28 @@ echo "testBuildFailure: Original builder produced exit code: $r"
 # -----------------------------------------
 # Write the build log to the default output
 #
-# # from stdenv setup.sh
+# # based on stdenv setup.sh
+if [ -f .attrs.sh ] || [[ -n "${NIX_ATTRS_SH_FILE:-}" ]] || [[ -n "${NIX_ATTRS_JSON_FILE:-}" ]]; then
+    __structuredAttrs=1
+
+    # Source the attr-vars
+    if [ -e "${NIX_ATTRS_SH_FILE:-}" ]; then
+        . "$NIX_ATTRS_SH_FILE"
+    elif [ -f .attrs.sh ]; then
+        . .attrs.sh
+    fi
```

Comment:
Is this reliably setup before running this script?

```suggestion
    if [ -e "$NIX_ATTRS_SH_FILE" ]; then
        . "$NIX_ATTRS_SH_FILE"
    fi
```

----

Or even?
```suggestion
    . "$NIX_ATTRS_SH_FILE"
```